### PR TITLE
Refactor kubeflow-katib for python subpackages to utilize venv

### PR DESCRIPTION
**Pull request by srishtih**
Fixes:

Related:

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: kubeflow-katib.yaml [hunk 1]
```diff
@@ -97,64 +97,55 @@ subpackages:
...
-        - numpy
-        - py3-scikit-learn
-        - py3-scipy
-        - py3-forestci
-        - py3-protobuf
-        - py3-googleapis-common-protos
-        - cython
+        - python-3.11-dev
+        - py3.11-pip
         - libgomp
```
Comment (developer-guy) [inline kubeflow-katib.yaml:102]: we should change it with `openmp-17`

File: kubeflow-katib.yaml [hunk 2]
```diff
@@ -67,165 +67,165 @@ subpackages:
...
           mv ./$SUGGESTION_DIR/* ${{targets.subpkgdir}}/opt/katib/$SUGGESTION_DIR
-
+  
   - name: katib-tfevent-metricscollector
     description: Kubeflow Katib tfevent-metricscollector
+    dependencies:
+      runtime:
+        - openmp-17
+        - python-3.11-dev
+        - py3.11-pip
```
Comment (developer-guy) [inline kubeflow-katib.yaml:90]: according to the error in CI, we should add `openmp-17` here

Comment (srishtih) [reply kubeflow-katib.yaml:90]: Done

**Comment by srishtih** [comment]
In remediation to the bug reported by CapitalOne https://github.com/chainguard-dev/customer-issues/issues/911
`ModuleNotFoundError: No module named 'imp'`

**Review by mamccorm — APPROVED** [review]

**Pull request merged**